static/js/components/EducationForm.js

```diff
@@ -0,0 +1,285 @@
+import React from 'react';
+import Button from 'react-mdl/lib/Button';
+import IconButton from 'react-mdl/lib/IconButton';
+import Grid, { Cell } from 'react-mdl/lib/Grid';
+import Switch from 'react-mdl/lib/Switch';
+import FABButton from 'react-mdl/lib/FABButton';
+import Icon from 'react-mdl/lib/Icon';
+import Dialog from 'material-ui/Dialog';
+import { Card } from 'react-mdl/lib/Card';
+import _ from 'lodash';
+import moment from 'moment';
+
+import ProfileFormFields from '../util/ProfileFormFields';
+import { generateNewEducation } from "../util/util";
+import { saveAndContinue } from '../util/profile_edit';
+import { HIGH_SCHOOL } from '../constants';
+
+class EducationForm extends ProfileFormFields {
+  constructor(props) {
+    super(props);
+    this.educationLevelLabels = {};
+    this.educationLevelOptions.forEach(level => {
+      this.educationLevelLabels[level.value] = level.label;
+    });
+  }
+
+  static propTypes = {
+    profile:                        React.PropTypes.object,
+    ui:                             React.PropTypes.object,
+    updateProfile:                  React.PropTypes.func,
+    clearProfileEdit:               React.PropTypes.func,
+    errors:                         React.PropTypes.object,
+    setEducationDialogVisibility:   React.PropTypes.func,
+    setEducationDialogIndex:        React.PropTypes.func,
+    setEducationDegreeLevel:        React.PropTypes.func,
+    setEducationDegreeInclusions:   React.PropTypes.func,
+  };
+
+  openNewEducationForm = (level, index) => {
+    const {
+      profile,
+      updateProfile,
+      setEducationDialogIndex,
+      setEducationDegreeLevel,
+      setEducationDialogVisibility,
+    } = this.props;
+    let newIndex = index;
+    if (index === null){
+      newIndex = profile['education'].length;
+    }
+    /* add empty education */
+    let clone = Object.assign({}, profile);
```

Comment:
We use `clone` for similar temporary objects elsewhere in the codebase, and I think given that we're only using it on three lines it's clear what's going on. I'd prefer to stick with the naming convention we're using.